src/replayToMovie.js

```diff
@@ -81,6 +79,46 @@ const p5Inst = new P5(function (p5obj) {
   p5obj.height = HEIGHT;
 });
 
+createNewP5RenderingCanvas();
+
+const backgroundEffects = new Effects(p5Inst, 1);
+const foregroundEffects = new Effects(p5Inst, 0.8);
+
+/**
+ * Create an emulated canvas for P5 to use as a rendering context.
+ *
+ * We have to create a new canvas on every request, otherwise under periods of
+ * high traffic the canvas can get into a state where it "freezes" and repeats
+ * a single frame for the length of an entire video.
+ *
+ * See https://github.com/code-dot-org/dance-party/issues/514 for more context
+ */
+function createNewP5RenderingCanvas() {
+  const canvas = window.document.createElement('canvas');
+  p5Inst._renderer = new P5.Renderer2D(canvas, p5Inst, false);
+  p5Inst._renderer.resize(WIDTH, HEIGHT);
+
+  // stub p5.scale to prevent any attempt at scaling down to 0, since that
+  // breaks node canvas (even though it works fine in the browser). Instead
+  // just scale down to something really, really small.
+  //
+  // See https://github.com/Automattic/node-canvas/issues/702
+  const origScale = p5Inst._renderer.scale;
+  p5Inst._renderer.scale = function(x, y) {
+    if (x === 0) {
+      x = 0.001;
+    }
+
+    if (y === 0) {
+      y = 0.001;
+    }
+
+    return origScale.call(this, x, y);
```

Comment:
Note: Hopefully irrelevant to our use case, but [`p5.scale()`](https://p5js.org/reference/#/p5/scale) supports a few other call patterns we should make sure we don't depend on anywhere.

- `scale([x, y])` is probably not common, but may be worth spot-checking Effects.js.
- `scale(createVector(x, y))` probably a non-issue for us.
- `scale(x, y, z)` should only be relevant when using WebGL so not an issue for us.

Reply:
Alternatively, I could apply this override to [canvas.scale()](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/scale) instead. I had been considering doing that anyway